aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -1227,6 +1251,24 @@ public <T extends DynamoDbClientBuilder> void applyDynamoDbEndpointConfiguration
     configureEndpoint(builder, dynamoDbEndpoint);
   }
 
+  public Region restSigv4SigningRegion() {
+    if (restSigv4SigningRegion == null) {
+      this.restSigv4SigningRegion =
+          DefaultAwsRegionProviderChain.builder().build().getRegion().id();
```

Comment:
Lazily loading the default region so that we don't fail if it cannot be resolved in the current environment.